src/Conversion/ONNXToKrnl/Tensor/GatherND.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- GatherND.cpp - Lowering GatherND Op -----------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX GatherND Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+#include "llvm/Support/Debug.h"
+#include <numeric>
+
+#define DEBUG_TYPE "gather_nd_onnx_to_krnl"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+struct ONNXGatherNDOpLowering : public ConversionPattern {
+  ONNXGatherNDOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(
+            typeConverter, ONNXGatherNDOp::getOperationName(), 1, ctx) {}
+
+  // When true causes injection of print stmts in the generated code.
+  static constexpr bool emitPrintStmts = false;
+
+  static void printIndices(
```

Comment:
Could you comment what this function is for? Ok its for debugging, just explain.

Reply:
done